packages/orbit-components/src/BadgeList/BadgeListItem/index.tsx

```diff
@@ -30,57 +30,43 @@ export const getIconColor = (type: Type) => {
   return type;
 };
 
-export const StyledBadgeListItem = styled.li`
-  ${({ theme }) => css`
-    display: flex;
-    flex-direction: row;
-    width: 100%;
-    & + & {
-      margin-top: ${theme.orbit.spaceXXSmall};
-    }
-  `};
-`;
+export const StyledBadgeListItem = ({ children, ...props }) => (
+  <li className="[&_+_&]:mt-xxs flex w-full flex-row" {...props}>
+    {children}
+  </li>
+);
 
-StyledBadgeListItem.defaultProps = {
-  theme: defaultTheme,
-};
-
-export const StyledVerticalBadge = styled.div<{ $type?: Props["type"] }>`
-  ${({ theme, $type }) => css`
-    background: ${getBackground({ theme, $type })};
-    display: flex;
-    align-items: center;
-    justify-content: center;
-    margin-${right}: ${theme.orbit.spaceXSmall};
-    flex-shrink: 0;
-    height: ${theme.orbit.heightIconLarge};
-    width: ${theme.orbit.widthIconLarge};
-    border-radius: ${theme.orbit.borderRadiusCircle};
-    svg {
-      height: ${theme.orbit.heightIconSmall};
-      width: ${theme.orbit.widthIconSmall};
-    }
-  `};
-`;
-
-StyledVerticalBadge.defaultProps = {
-  theme: defaultTheme,
+export const StyledVerticalBadge = ({
+  children,
+  $type,
+  ...props
+}: {
+  children: React.ReactNode;
+  $type: Props["type"];
+}) => {
+  const theme = useTheme();
+  return (
+    <div
+      className={cx(
+        "me-xs h-icon-large w-icon-large rounded-circle flex flex-shrink-0 items-center justify-center",
+        "[&_svg]:h-icon-small [&_svg]:w-icon-small",
+      )}
+      style={{ background: getBackground({ theme, $type }) ?? undefined }}
+      {...props}
+    >
+      {children}
+    </div>
+  );
 };
 
-export const StyledBadgeContent = styled.div`
-  ${({ theme }) => css`
-    display: inline-flex;
-    align-items: center;
-
-    .orbit-tooltip-wrapper .orbit-text {
-      font-weight: ${theme.orbit.fontWeightMedium};
-    }
-  `};
-`;
-
-StyledBadgeContent.defaultProps = {
-  theme: defaultTheme,
-};
+export const StyledBadgeContent = ({ children, ...props }) => (
```

Comment:
This requires more work to keep compatibility in `ItineraryBadgeListItem`. The usage of this component there uses the `css` prop, that is accepted by styled components, but not here… So we have two options here:
- we open the style prop and spread it AND in `ItineraryBadgeListItem` we adjust accordingly
OR
- we create a new prop that applies that style that `ItineraryBadgeListItem` needs and we add that prop the component usage there

I'm fine with any approach. And the same suggestion regarding naming.

Reply:
I didn't even realised of the `css` prop, thanks for the thorough check.

I've created a `style` prop in `BadgeContent` and adapted `ItineraryBadgeListItem` to pass the values with not styled-components' `css` function. Is this what you had in mind?